### DISCUSSION TITLE
register_uri_scheme_protocol not working

### DISCUSSION THREAD

**Discussion by sazzer** [tauri-apps/tauri#5900] [Q&A]
I'm trying to implement an OAuth2 flow, whereby I'm opening a new window to start the authentication and then handling the callback to get an access token.

The first part of that was remarkably easy. The second part, not so much so far. We've got the redirect_uri set to a custom HTTP scheme so that it doesn't go and call the wrong service by accident. As best as I understand it, `register_uri_scheme_protocol` means I can then handle calls to this scheme from my webview, react to the details from the request and all is good.

Except that my handler isn't getting called and I'm getting a popup error: "There is no application set to open the URL .....".

Am I missing something about how to get my window to handle this?

Roughly speaking my code is:
```
fn main() {
    tauri::Builder::default()
        .invoke_handler(tauri::generate_handler![login])
        .register_uri_scheme_protocol("x-callback", callback)
        .run(tauri::generate_context!())
        .expect("error while running tauri application");
}

#[tauri::command]
pub async fn login(app: tauri::AppHandle) {
    let uri = build_auth_url();

    tauri::WindowBuilder::new(
        &app,
        "oauth2",
        tauri::WindowUrl::External(uri.parse().unwrap()),
    )
    .build()
    .unwrap();
}

pub fn callback(
    app: &tauri::AppHandle,
    req: &Request,
) -> Result<Response, Box<dyn std::error::Error>> {
    tracing::info!("Callback URI: {}", req.uri());

    app.get_window("oauth2").unwrap().close().unwrap();
    todo!()
}
```

Cheers

**Comment by mail4metablocks** [comment]
It looks like the issue you are experiencing may be due to the fact that the custom URI scheme you have registered ("x-callback") is not being handled by any application on your system.

To resolve this issue, you can try the following:

    Make sure that your Tauri app is registered as the default handler for the "x-callback" URI scheme. You can do this by calling the register_uri_scheme_protocol method with the is_default parameter set to true. For example:

tauri.register_uri_scheme_protocol("x-callback", callback, true);

    If the above step does not work, you can try manually registering your Tauri app as the default handler for the "x-callback" URI scheme on your system. This will typically involve modifying the system's registry or configuration files to associate the "x-callback" URI scheme with your Tauri app. The exact steps for doing this will depend on your operating system and may require some additional research.
